[bug] `string_decoder/` is not builtin

### CJS & ESM
CJS and ESM both think `string_decoder/` is [external](https://www.npmjs.com/package/string_decoder)

```txt
> require('string_decoder/')
Uncaught Error: Cannot find module 'string_decoder/'
```

```
> await import('string_decoder/')
Uncaught:
Error [ERR_MODULE_NOT_FOUND]:
```

### Actual
```txt
> isBuiltinModule = require('is-builtin-module');
> isBuiltinModule('string_decoder/')
true
```

### Expected
```txt
> isBuiltinModule = require('is-builtin-module');
> isBuiltinModule('string_decoder/')
false
```

### Related
```txt
> isBuiltinModule('string_decoder')
true
> isBuiltinModule('fs/promises')
true
```

--- a/index.js
+++ b/index.js
@@ -14,7 +14,7 @@ module.exports = moduleName => {
 	}
 
 	const slashIndex = moduleName.indexOf('/');
-	if (slashIndex !== -1) {
+	if (slashIndex !== -1 && slashIndex !== moduleName.length - 1) {
 		moduleName = moduleName.slice(0, slashIndex);
 	}
 

--- a/test.js
+++ b/test.js
@@ -4,12 +4,12 @@ import isBuiltinModule from '.';
 test('main', t => {
 	t.true(isBuiltinModule('fs'));
 	t.true(isBuiltinModule('console'));
+	t.true(isBuiltinModule('punycode'));
 
 	t.true(isBuiltinModule('fs/promises'));
 	t.true(isBuiltinModule('assert/strict'));
 
 	// These are actually not, but should not exist
-	t.true(isBuiltinModule('fs/'));
 	t.true(isBuiltinModule('fs/unknown'));
 	t.true(isBuiltinModule('fs/promises/unknown'));
 	t.true(isBuiltinModule('fs/promises?query=1'));
@@ -17,6 +17,7 @@ test('main', t => {
 	t.true(isBuiltinModule('node:fs'));
 	t.true(isBuiltinModule('node:fs/promises'));
 
+	t.false(isBuiltinModule('punycode/'));
 	t.false(isBuiltinModule('unicorn'));
 	t.false(isBuiltinModule('unknown'));
 	t.false(isBuiltinModule('FS'));

